src/sentry/integrations/vsts/webhooks.py

```diff
@@ -89,15 +96,110 @@ def handle_status_change(self, integration, external_issue_key,
             }
             installation.sync_status_inbound(external_issue_key, data)
 
-    def parse_email(self, email):
-        return EMAIL_PARSER.search(email).group(1)
 
-    def create_subscription(self, instance, identity_data, oauth_redirect_url, external_id):
-        client = self.get_client(Identity(data=identity_data), oauth_redirect_url)
-        shared_secret = self.create_webhook_secret()
-        return client.create_subscription(instance, external_id, shared_secret), shared_secret
+class PullRequestWebhook(Webhook):
+    def __call__(self, data, integration):
+        organization_ids = OrganizationIntegration.objects.filter(
+            integration_id=integration.id,
+        ).values_list('organization_id', flat=True)
+        for organization_id in organization_ids:
+            installation = integration.get_installation(organization_id)
+            try:
+                repo = Repository.objects.get(
+                    organization_id=organization_id,
+                    provider=PROVIDER_NAME,
+                    external_id=six.text_type(data['resource']['repository']['id']),
+                )
+            except Repository.DoesNotExist as e:
+                installation.logger.error(
+                    e,
+                    extra={
+                        'integration_id': installation.model.id,
+                        'organization_id': organization_id,
+                    }
+                )
+                continue
+            author = self.get_or_create_author(data, installation, organization_id)
+            self.create_pull_request(data, repo, author, organization_id)
+
+    def get_or_create_author(self, data, installation, organization_id):
+        author = data['resource']['createdBy']
+        author_email = author['uniqueName']
+        author_id = author['id']
+        author_name = author['displayName']
+        try:
+            commit_author = CommitAuthor.objects.get(
+                external_id=author_id,
+                organization_id=organization_id,
+                email=author_email,
```

Comment:
I think we should exclude `email` from this lookup.

`CommitAuthor` has a unique constraint on `(external_id, organization_id)`. If the person has changed their email in VSTS, this `get()` won't find anything and we'll attempt to create a new `CommitAuthor` with the same `external_id` and `organization_id`, but a different `email`. Pretty sure it'll blow up because of the constraint in that case.

Excluding `email` will find the corresponding `CommitAuthor`, it'll just have the "wrong" email. We can either update the email or just ignore the new address. Not sure what the right call there is, though.